.github/workflows/ci.yml

```
@@ -405,7 +304,6 @@ jobs:
   ##############################################################################
 
   test-rust-matrix:
-    needs: rust-prebuild-matrix
     strategy:
```

```suggestion
    env:
         CARGO_PROFILE: ci-rust
    strategy:
```

Missing to use the custom rust profile for the rust job

---

This is not needed given we don't do `poetry install` here.

On top of that this env variable is misleading given it may let you think `cargo build` will take it into account ([as far as I know this is not the case](https://doc.rust-lang.org/cargo/reference/environment-variables.html), but there is pretty similar env var such as `CARGO_PROFILE_<name>_BUILD_OVERRIDE_<key>` which make it even more misleading ^^)

In the end I find it far more readable to be explicit by avoiding global environ variable here

---

So wha't the goal of having the profile `ci-rust` if we don't use it in the ci :thinking:

---

It is used in the CI to build the rust tests

---

Yes I've missed it